astropy/io/fits/fitstime.py

```diff
@@ -127,33 +175,25 @@ def _verify_column_info(column_info, global_info):
     scale = column_info.get('TCTYP', None)
     unit = column_info.get('TCUNI', None)
     location = column_info.get('TRPOS', None)
-
-    if location is not None:
-
-        if location == 'TOPOCENTER':
-            column_info['location'] = EarthLocation(global_info['OBSGEO-X'],
-                                                    global_info['OBSGEO-Y'],
-                                                    global_info['OBSGEO-Z'],
-                                                    unit='m')
-        else:
-            column_info['location'] = None
+    global_ref_pos = global_info.get('TREFPOS', None)
 
     if scale is not None:
 
-        if scale.lower() in Time.SCALES:
+        # Non-linear coordinate types have "4-3" form and are not time coordinates
+        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
```

Comment:
This is indirectly tested when TCTYP = 'RA---TAN', which is not a time column. The chandra file has these spatial coordinates.